sympy/physics/mechanics/kane.py

```diff
@@ -132,62 +177,96 @@ def __init__(self, frame, q_ind, u_ind, kd_eqs=None, q_dependent=None,
 
     def _initialize_vectors(self, q_ind, q_dep, u_ind, u_dep, u_aux):
-        """Initialize the coordinate and speed vectors."""
+        """Initialize the coordinate and speed vectors and sets the private
+        class attributes that contain them. The results are publicly accessible
+        through the attributes described in the class docstring.
 
-        none_handler = lambda x: Matrix(x) if x else Matrix()
+        """
 
         # Initialize generalized coordinates
-        q_dep = none_handler(q_dep)
-        if not iterable(q_ind):
-            raise TypeError('Generalized coordinates must be an iterable.')
+        q_dep = _none_handler(q_dep)
         if not iterable(q_dep):
             raise TypeError('Dependent coordinates must be an iterable.')
+        if not iterable(q_ind):
+            raise TypeError('Generalized coordinates must be an iterable.')
         q_ind = Matrix(q_ind)
+
+        self._qind = q_ind
         self._qdep = q_dep
         self._q = Matrix([q_ind, q_dep])
         self._qdot = self.q.diff(dynamicsymbols._t)
 
         # Initialize generalized speeds
-        u_dep = none_handler(u_dep)
-        if not iterable(u_ind):
-            raise TypeError('Generalized speeds must be an iterable.')
+        u_dep = _none_handler(u_dep)
         if not iterable(u_dep):
             raise TypeError('Dependent speeds must be an iterable.')
+        if not iterable(u_ind):
+            raise TypeError('Generalized speeds must be an iterable.')
         u_ind = Matrix(u_ind)
+
+        self._uind = u_ind
         self._udep = u_dep
         self._u = Matrix([u_ind, u_dep])
         self._udot = self.u.diff(dynamicsymbols._t)
-        self._uaux = none_handler(u_aux)
+        self._uaux = _none_handler(u_aux)
+
+        self._n = len(self._u)
+        self._m = len(u_dep)
+        self._o = len(q_dep)
+        self._p = len(u_ind)
 
     def _initialize_constraint_matrices(self, config, vel, acc):
-        """Initializes constraint matrices."""
+        """Initializes constraint matrices.
 
-        # Define vector dimensions
-        o = len(self.u)
-        m = len(self._udep)
-        p = o - m
-        none_handler = lambda x: Matrix(x) if x else Matrix()
+        Given::
+
+            config = f_h(q, t) = 0
+            vel = K_nh(q, t) * u + f_nh(q, t) = 0
+            acc = K_dnh(q, t) * u' + f_dnh(q, u, t) = 0
+
+        this function finds and stores::
+
+            f_h, K_nh, f_nh, K_dnh, f_dnh, and A_rs
+
+        where::
+
+            u = [u_s, u_r]^T
+            u_r = A_rs * u_s + B_r
+
+        Parameters
+        ==========
+        config : iterable of SymPy expressions
+            The configuration constraint expressions.
+        vel : iterable of SymPy expressions
+            The velocity constraint expressions.
+        acc : iterable of SymPy expressions
+            The acceleration constraint expressions.
+
+        """
```

Comment:
This docstring is really well done and I like the inclusion of the equation definitions!